### DISCUSSION TITLE
Unique validation while updating a value with Extracted Form

### DISCUSSION THREAD

**Discussion by sayedsadat344** [livewire/livewire#6438] [1. Help] [answered]
I have extracted a form called `ProductForm` as follow:

```
  #[Rule('required|unique:products,name|min:5')]
    public $name = '';

    #[Rule('required|min:10')]
    public $description = '';
```

In Product component where we add/update

` public ProductForm $productForm;`

```
   public function updateProduct()
    {

        $this->validate();


        $prod = Product::find($this->product_id);


        $this->productForm->image ? $this->productForm->image->store('product/images', 'public') : $prod->image;

        $prod->update($this->productForm->all());

        $this->dispatch('alert', [
            'type' => 'success',
            'message' => "Product updated Successfully!!",
        ]);

        return $this->redirect('/products', navigate: true);

    }
```

Add new product works fine, but when we want to update the form it gives error that `"The name is already taken"`.
even though `I am not updating the name field.`

how to skip `unique validation` check for the current product? it should check in the rest of the table,

**Comment by N1ebieski** [comment]
https://github.com/livewire/livewire/discussions/6350#discussioncomment-6730354

**Reply by sayedsadat344** [reply, marked as answer]
It means the current V3 attribute `#[Rule('required|unique:products,name|min:5')] ` can not be used with this `ignore `functionality `Rule::unique('products', 'id')->ignore($this->id)` 

We have to overwrite it in old format of validation

```
public function rules(): array
{
    return [
        'name' => [
            'required',
            'string',
            'max:255',
            Rule::unique('products', 'id)->ignore($this->component->id)
        ],
   ];
}
```

So, what good is the new format for??

**Reply by PhiloNL** [reply]
Both approaches are valid ways to define rules. For simple validation, you can use attributes. You can use the `rules()` method when you have more complex validation rules. Not being able to use something like `Rule::unique('products', 'id')->ignore($this->id)` within the Rule attribute is a limitation of PHP.

**Comment by tobidosumu** [comment]
### There is a similar issue when using the Validate() in Livewire 3.

The issue is that the **unique** validation method is triggered upon submitting the update form whilst **email** is left un-changed.
This behaviour is undesired as a user may not want to update their email.

Please check the code below for further details:
#[Validate('unique:users,email', message: 'Sorry a user with this email already exists.')]

**Please note:** The message above is meant to show when a new user is being created. 
If the registered email already exists in the database, then this message is triggered.
